### PR TITLE
Deprecate existing Prometheus Feature 

### PR DESCRIPTION
**Pull request by Nipunich**
## Purpose
Deprecate existing Prometheus Feature and change module structure.

### CODE REVIEW

File: components/mediation/data-publishers/org.wso2.micro.integrator.observability/pom.xml [hunk 1]
```diff
@@ -66,8 +75,9 @@
                         <Bundle-Name>${project.artifactId}</Bundle-Name>
                         <Axis2Module>${project.artifactId}-${project.version}</Axis2Module>
                         <Export-Package>
-                            org.wso2.micro.integrator.prometheus.observability.* ;version="${project.version}"
+                            org.wso2.micro.integrator.observability.* ;version="${project.version}"
                         </Export-Package>
+                        <Private-Package>io.prometheus.client.*</Private-Package>
```
Comment (arunans23) [inline components/mediation/data-publishers/org.wso2.micro.integrator.observability/pom.xml:80]: Why are we private packaging here?

Comment (Nipunich) [reply components/mediation/data-publishers/org.wso2.micro.integrator.observability/pom.xml:80]: This was done to make the io.prometheus.client.* package to copy into the bundle but not export.
This was done to fix the issue com.sun package not found thrown during the build phase.

File: features/mediation-features/data-publisher-features/org.wso2.micro.integrator.observability.feature/pom.xml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>data-publisher-features</artifactId>
+        <groupId>org.wso2.ei</groupId>
+        <version>1.2.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>org.wso2.micro.integrator.observability.feature</artifactId>
```
Comment (arunans23) [inline features/mediation-features/data-publisher-features/org.wso2.micro.integrator.observability.feature/pom.xml:13]: Don't we need a name tag for the feature here?

File: features/mediation-features/data-publisher-features/org.wso2.micro.integrator.observability.feature/pom.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    <description>
+        This feature contains the core bundles required for publishing metrics data to Prometheus
```
Comment (arunans23) [inline features/mediation-features/data-publisher-features/org.wso2.micro.integrator.observability.feature/pom.xml:15]: We can change the description to reflect that we have a generic observability feature.

Comment (Nipunich) [reply features/mediation-features/data-publisher-features/org.wso2.micro.integrator.observability.feature/pom.xml]: Fixed

File: features/mediation-features/data-publisher-features/org.wso2.micro.integrator.observability.feature/pom.xml (hunk 2 above)
Comment (Nipunich) [reply features/mediation-features/data-publisher-features/org.wso2.micro.integrator.observability.feature/pom.xml:13]: Added the name tag.

File: features/mediation-features/data-publisher-features/org.wso2.micro.integrator.observability.feature/pom.xml [hunk 4]
```diff
@@ -0,0 +1,87 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Nirothipan) [inline features/mediation-features/data-publisher-features/org.wso2.micro.integrator.observability.feature/pom.xml:1]: Please add license header

File: components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/metric/publisher/MetricResource.java [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Nirothipan) [inline components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/metric/publisher/MetricResource.java:2]: we shouldn't be updating the year of existing files.

File: features/mediation-features/data-publisher-features/org.wso2.micro.integrator.observability.feature/pom.xml (hunk 4 above)
Comment (Nipunich) [reply features/mediation-features/data-publisher-features/org.wso2.micro.integrator.observability.feature/pom.xml:1]: Fixed

File: components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/metric/publisher/MetricResource.java (hunk 5 above)
Comment (Nipunich) [reply components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/metric/publisher/MetricResource.java]: Fixed

File: components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/metric/publisher/MetricAPI.java [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sdkottegoda) [inline components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/metric/publisher/MetricAPI.java:2]: updating the license of an older file

**Pull request merged**